src/lib/y2partitioner/actions/delete_partitions.rb

```diff
@@ -39,7 +39,7 @@ def initialize(*args)
       def delete
         log.info "deleting partitions from #{device}"
 
-        device.partitions.each { |p| device.partition_table.delete_partition(p) }
```

Comment:
good catch. So it delete extended and then it tries to delete logical which is no longer there.